plugins/plugin-chart-table/src/types.ts

```diff
@@ -25,52 +25,49 @@ import {
   DataRecord,
   DataRecordValue,
   DataRecordFilters,
-  QueryData,
+  GenericDataType,
+  QueryMode,
+  ChartDataResponseResult,
+  QueryFormData,
 } from '@superset-ui/core';
 
-export enum DataType {
-  Number = 'number',
-  String = 'string',
-  DateTime = 'datetime',
-}
-
 export type CustomFormatter = (value: DataRecordValue) => string;
 
 export interface DataColumnMeta {
   // `key` is what is called `label` in the input props
   key: string;
   // `label` is verbose column name used for rendering
   label: string;
-  dataType: DataType;
+  dataType: GenericDataType;
   formatter?: TimeFormatter | NumberFormatter | CustomFormatter;
 }
 
-export interface TableChartData<D extends DataRecord = DataRecord> {
-  records: D[];
+export interface TableChartData {
+  records: DataRecord[];
   columns: string[];
 }
 
-export interface TableChartFormData {
-  alignPn?: boolean;
-  colorPn?: boolean;
-  includeSearch?: boolean;
-  pageLength?: string | number | null; // null means auto-paginate
+export type TableChartFormData = QueryFormData & {
```

Comment:
The benefit of not doing case transformation is that we can directly reuse this typing for both `formData` sent to the API and received by the chart renderer (e.g. in both `buildQuery` and `transformProps`.)